weave/tests/trace/test_client_cost.py

```diff
@@ -0,0 +1,209 @@
+    assert len(res) == 2
+    if res[0]["llm_id"] == "my_model_to_delete":
+        assert res[0]["completion_token_cost_unit"] == "USD"
+        assert res[1]["completion_token_cost_unit"] == "doubleoons"
+    else:
+        assert res[0]["completion_token_cost_unit"] == "doubleoons"
+        assert res[1]["completion_token_cost_unit"] == "USD"
+
+    # Add another cost of the same llm_id
+    costs = {
+        "my_model_to_delete3": {
+            "prompt_token_cost": 500,
+            "completion_token_cost": 1000,
+            "effective_date": datetime(1998, 10, 3),
+        },
+    }
+    res = client.server.cost_create(
+        tsi.CostCreateReq(
+            project_id=project_id,
+            costs=costs,
+            wb_user_id="VXNlcjo0NTI1NDQ=",
+        )
+    )
+
+    assert len(res.ids) == 1
+    last_id = res.ids[0][0]
+
+    # query multiple costs by llm_id
+    res = client.query_costs(llm_ids=["my_model_to_delete3"])
+    assert len(res) == 2
+    if res[0]["effective_date"].year == 2021:
+        assert res[1]["effective_date"].year == 1998
+    else:
+        assert res[1]["effective_date"].year == 2021
+        assert res[0]["effective_date"].year == 1998
+
+    # query with limit and offset
+    res = client.query_costs(llm_ids=["my_model_to_delete3"], limit=1)
+    assert len(res) == 1
+
+    res = client.query_costs(llm_ids=["my_model_to_delete3"], offset=1)
+    assert len(res) == 1
+
+    # query with query
+    res = client.query_costs(
+        query=Query(
+            **{
+                "$expr": {
+                    "$gt": [
+                        {"$getField": "completion_token_cost"},
+                        {"$literal": 25},
+                    ],
+                }
+            }
+        )
+    )
+    assert len(res) == 3
+
+    # purge costs
+    client.purge_costs(cost_ids[0][0])
+    res = client.query_costs()
+    assert len(res) == 4
+
+    client.purge_costs(cost_ids[0][0])
+    res = client.query_costs()
+    assert len(res) == 4
+
+    client.purge_costs([id[0] for id in cost_ids])
+    res = client.query_costs()
+    assert len(res) == 1
+    assert res[0]["id"] == last_id
+
+    # purge last cost
+    client.purge_costs(last_id)
+    res = client.query_costs()
+    assert len(res) == 0
+
+
+def test_purge_only_ids(client):
+    is_sqlite = isinstance(client.server._internal_trace_server, SqliteTraceServer)
+    if is_sqlite:
+        # only run this test for sqlite
+        return
+
+    project_id = client._project_id()
+    costs = {
+        "my_model_to_delete": {
+            "prompt_token_cost": 5,
+            "completion_token_cost": 10,
+        },
+    }
+    # Create some costs
+    res = client.server.cost_create(
+        tsi.CostCreateReq(
+            project_id=project_id,
+            costs=costs,
+            wb_user_id="VXNlcjo0NTI1NDQ=",
+        )
+    )
+
+    cost_ids = res.ids
+    assert len(cost_ids) == 1
+
+    with pytest.raises(InvalidRequest):
```

Comment:
We could try to hit every `InvalidRequest` raise here, codecov

Reply:
let me add a test for the validation utils